src/strand.rs

```diff
@@ -0,0 +1,306 @@
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Ord, PartialOrd, Copy)]
+pub enum NoStrand {
+    Unknown,
+}
+
+impl Same for NoStrand {
+    fn same(&self, _s1: &Self) -> bool {
+        true
+    }
+}
+
+impl Display for NoStrand {
+    fn fmt(&self, _f: &mut Formatter) -> fmt::Result {
+        Ok(())
+    }
+}
+
+impl Strandedness for NoStrand {
+    fn reverse(&self) -> Self {
+        NoStrand::Unknown
+    }
+    fn try_req_strand(&self) -> Option<ReqStrand> {
+        None
+    }
+    fn strand_symbol(&self) -> &str {
+        "."
+    }
+    fn break_pos_strand(posstr: &str) -> Option<(&str, Self)> {
+        Some((posstr, NoStrand::Unknown))
+    }
+}
+
+/// Equality-like operator for comparing strand information. Unknown
+/// strands are not equal, but they are the "same" as other unknown
+/// strands.
+pub trait Same {
+    /// Indicate when two strands are the "same" -- two
+    /// unknown/unspecified strands are the "same" but are not equal.
+    fn same(&self, &Self) -> bool;
+}
+
+impl<T> Same for Option<T>
+where
+    T: Same,
+{
+    fn same(&self, s1: &Self) -> bool {
+        match (self, s1) {
+            (&Option::None, &Option::None) => true,
+            (&Option::Some(ref x), &Option::Some(ref x1)) => x.same(x1),
+            (_, _) => false,
+        }
+    }
+}
+
+/// Strand information for annotations on sequences
+pub trait Strandedness {
+    /// Reversed strand
+    fn reverse(&self) -> Self;
+
+    /// Convert into a `ReqStrand`, indicating a specific, known
+    /// strand. The return is wrapped in an `Option`, so unknown
+    /// strands will return `None`.
+    fn try_req_strand(&self) -> Option<ReqStrand>;
+
+    /// Symbol denoting the strand. By convention, in BED and GFF
+    /// files, the forward strand is `+`, the reverse strand is `-`,
+    /// and unknown or unspecified strands are `.`.
+    fn strand_symbol(&self) -> &str;
+
+    /// Parse a strand designator from the end of a location display
+    /// string.
+    ///
+    /// Strand information can be represented by a `(+)` or `(-)` at
+    /// the end of a display string. This function extracts that
+    /// strand information if it's present, and returns a pair
+    /// comprising the rest of the string and the parsed strand.
+    ///
+    /// When strand information is required but it isn't found in the
+    /// display string, then `None` is returned.
+    ///
+    /// When strand information is optional and it isn't found, then
+    /// the entire input string is returned along with the unknown
+    /// strand designation.
+    fn break_pos_strand(&str) -> Option<(&str, Self)>
+    where
+        Self: marker::Sized;
+}
+
+/// Strand information when the strand is guaranteed to be know.
+pub trait KnownStrandedness: Strandedness {
+    /// Convert into a `ReqStrand`, indicating a specific, known
+    /// strand. As strandedness is guaranteed, this cannot fail.
+    fn req_strand(&self) -> ReqStrand;
+
+    fn product<S>(&self, s: S) -> S
```

Comment:
Please add doc comments to all functions.

Reply:
Trait was removed...